### PR TITLE
test: add tests for GithubStore

### PR DESCRIPTION
**Pull request by umglurf**
Solves #42

### CODE REVIEW

File: pkg/store/github/github_test.go [hunk 1]
```diff
@@ -4,8 +4,192 @@
...
+		c := github.NewClient(mockedHTTPClient)
+		store := &GitHubStore{
+			ownerFilter: "test-owner",
+			topicFilter: "test-topic",
+			client:      c,
+			cache:       make(map[string][]*core.ModuleVersion),
+		}
+		store.client = c
+		err := store.ReloadCache(context.Background())
+		is.True(err != nil)
```
Comment (stigok) [inline pkg/store/github/github_test.go:66]: `is.NoErr(err)`

File: pkg/store/github/github_test.go [hunk 2]
```diff
@@ -4,8 +4,192 @@
...
+		c := github.NewClient(mockedHTTPClient)
+		store := &GitHubStore{
+			ownerFilter: "test-owner",
+			topicFilter: "test-topic",
+			client:      c,
+			cache:       make(map[string][]*core.ModuleVersion),
+		}
+
+		err := store.ReloadCache(context.Background())
+		is.True(err == nil)
```
Comment (stigok) [inline pkg/store/github/github_test.go:40]: `is.NoErr(err)`

File: pkg/store/github/github_test.go [hunk 3]
```diff
@@ -4,8 +4,192 @@
...
+	c := github.NewClient(mockedHTTPClient)
+	store := &GitHubStore{
+		ownerFilter: "test-owner",
+		topicFilter: "test-topic",
+		client:      c,
+		cache:       make(map[string][]*core.ModuleVersion),
+	}
+
+	err := store.ReloadCache(context.Background())
+	if err != nil {
```
Comment (stigok) [inline pkg/store/github/github_test.go:112]: Comments on same line are printed to stderr along with the (failed) test results by `is`
```
is.NoErr(err) // Could not ReloadCache
```

File: pkg/store/github/github_test.go [hunk 4]
```diff
@@ -4,8 +4,192 @@
...
+		ver, err := store.GetModuleVersion(context.Background(), "test-owner", "test-repo", "generic", "1.0.0")
+		is.True(err == nil)
+		is.Equal(ver.Version, "1.0.0")
+		is.Equal(ver.SourceURL, "git::ssh://git@github.com/test-owner/test-repo.git?ref=v1.0.0")
+	})
+
+	t.Run("errs when missing", func(t *testing.T) {
+		is := is.New(t)
+		ver, err := store.GetModuleVersion(context.Background(), "test-owner", "test-repo", "generic", "1.0.1")
+		is.True(err != nil)
```
Comment (stigok) [inline pkg/store/github/github_test.go:127]: `is.NoErr(err)`

File: pkg/store/github/github_test.go [hunk 5]
```diff
@@ -4,8 +4,192 @@
...
+		is.Equal(len(versions), 3)
+		is.Equal(versions[0].Version, "1.0.0")
+		is.Equal(versions[1].Version, "1.0.1")
+		is.Equal(versions[2].Version, "2.0.0")
+	})
+
+	t.Run("errs when missing", func(t *testing.T) {
+		is := is.New(t)
+		versions, err := store.ListModuleVersions(context.Background(), "wrong", "wrong", "wrong")
+		is.True(err != nil)
```
Comment (stigok) [inline pkg/store/github/github_test.go:191]: `is.NoErr(err)`

File: pkg/store/github/github_test.go (hunk 3 above)
Comment (stigok) [reply pkg/store/github/github_test.go:112]: More of a FYI -- not important at all

**Review by stigok — APPROVED** [review]
Super nice to get tests for this!

File: pkg/store/github/github_test.go (hunk 3 above)
Comment (umglurf) [reply pkg/store/github/github_test.go]: changed

File: pkg/store/github/github_test.go (hunk 1 above)
Comment (umglurf) [reply pkg/store/github/github_test.go]: it's actually testing that we have an error

File: pkg/store/github/github_test.go (hunk 3 above)
Comment (umglurf) [reply pkg/store/github/github_test.go:112]: did not want to create an is instance on the outside, hence the if statement

File: pkg/store/github/github_test.go (hunk 4 above)
Comment (umglurf) [reply pkg/store/github/github_test.go:127]: it's checking that we have an error

**Review by stigok — APPROVED** [review]

**Pull request merged**